SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -107,55 +299,41 @@ void ASpatialDebugger::Tick(float DeltaSeconds)
 		{
 			LocalPlayerState = LocalPawn->GetPlayerState();
 		}
-
-		if (LocalPawn.IsValid())
-		{
-			SCOPE_CYCLE_COUNTER(STAT_SortingActors);
-			const FVector& PlayerLocation = LocalPawn->GetActorLocation();
-
-			EntityActorMapping.ValueSort([PlayerLocation](const TWeakObjectPtr<AActor>& A, const TWeakObjectPtr<AActor>& B) {
-				return FVector::Dist(PlayerLocation, A->GetActorLocation()) > FVector::Dist(PlayerLocation, B->GetActorLocation());
-			});
-		}
 	}
 }
 
-void ASpatialDebugger::BeginPlay()
+void ASpatialDebugger::OnEntityAdded(AActor* Actor)
 {
-	Super::BeginPlay();
-
-	check(NetDriver != nullptr);
-
-	if (!NetDriver->IsServer())
+	// Each client will only receive a PlayerController once.
+	if (Actor->IsA<APlayerController>())
 	{
-		EntityActorMapping.Reserve(ENTITY_ACTOR_MAP_RESERVATION_COUNT);
-
-		LoadIcons();
-
-		TArray<Worker_EntityId_Key> EntityIds;
-		NetDriver->StaticComponentView->GetEntityIds(EntityIds);
+		LocalPlayerController = Cast<APlayerController>(Actor);
 
-		// Capture any entities that are already present on this client (ie they came over the wire before the SpatialDebugger did).
-		for (const Worker_EntityId_Key EntityId : EntityIds)
+		if (GetNetMode() == NM_Client)
 		{
-			OnEntityAdded(EntityId);
+			LocalPlayerController->InputComponent->BindKey(ConfigUIToggleKey, IE_Pressed, this, &ASpatialDebugger::OnToggleConfigUI)
+				.bConsumeInput = false;
+			LocalPlayerController->InputComponent->BindKey(SelectActorKey, IE_Pressed, this, &ASpatialDebugger::OnSelectActor)
+				.bConsumeInput = false;
+			LocalPlayerController->InputComponent->BindKey(HighlightActorKey, IE_Pressed, this, &ASpatialDebugger::OnHighlightActor)
+				.bConsumeInput = false;
 		}
+	}
+}
 
-		FontRenderInfo.bClipText = true;
-		FontRenderInfo.bEnableShadow = true;
+void SpatialDebuggerSystem::ActorAuthorityIntentChanged(Worker_EntityId EntityId, VirtualWorkerId NewIntentVirtualWorkerId) const
```

Comment:
The layout in this cpp file mixes SpatialDebugger and SpatialDebuggerSystem. Ideally each class has it's own cpp file.

Reply:
Should I separate headers as well then?

Reply:
Yes if it's not much hassle!